src/parse.c

```diff
@@ -1437,31 +1442,43 @@ handle_wifi_access_points(NetplanParser* npp, yaml_node_t* node, const char* key
                 continue;
         }
 
+        ssid = scalar(key);
+
+        /* Skip access-points that already exist in the netdef */
+        if (npp->current.netdef->access_points && g_hash_table_contains(npp->current.netdef->access_points, ssid))
+            continue;
+
+        /* Check if there's already an SSID with that name in the list of APs we are parsing */
+        if (g_hash_table_contains(access_points, ssid)) {
```

Comment:
nitpick: This is about the SSIDs defined inside the YAML file being parsed currently, IIUC. Maybe the comment could be clarified in that regard.